packages/core/src/new-api/internal/execution/onchain-state-transitions.ts

```diff
@@ -595,6 +600,12 @@ async function checkTransactionComplete(
     receipt: ethers.providers.TransactionReceipt
   ) => ExecutionSuccess | ExecutionFailure | TransactionMessage
 ): Promise<OnchainStateTransitionContinue | OnchainStateTransitionPause> {
+  state.transactionLookupTimer.registerStartTime({
```

Comment:
Every time we check for a transaction by receipt we register a start time (this is idempotent, on the second time we ignore) with the `transactionLookupTimer` which encapsulates this ephemeral state.

Reply:
maybe `registerStartTimeIfNeeded`?